server/athenian/api/precompute/accounts.py

```diff
@@ -35,175 +35,191 @@
 
 async def main(context: PrecomputeContext, args: argparse.Namespace) -> None:
     """Precompute several accounts."""
-    log, sdb, mdb, pdb, rdb, cache, slack = (
-        context.log, context.sdb, context.mdb, context.pdb, context.rdb, context.cache,
-        context.slack,
+    time_to = datetime.combine(
+        date.today() + timedelta(days=1), datetime.min.time(), tzinfo=timezone.utc,
     )
-    successful_accounts = set()
-    failed_accounts = set()
-    time_to = datetime.combine(date.today() + timedelta(days=1),
-                               datetime.min.time(),
-                               tzinfo=timezone.utc)
     no_time_from = datetime(1970, 1, 1, tzinfo=timezone.utc)
     time_from = (time_to - timedelta(days=365 * 2)) if not os.getenv("CI") else no_time_from
     accounts = [int(p) for s in args.account for p in s.split()]
-    reposets = await sdb.fetch_all(
-        select([RepositorySet])
-        .where(and_(RepositorySet.name == RepositorySet.ALL,
-                    RepositorySet.owner_id.in_(accounts))))
-    reposets = [RepositorySet(**r) for r in reposets]
-    log.info("Heating %d reposets", len(reposets))
+    reposets = await _get_reposets(context.sdb, accounts)
+    context.log.info("Heating %d reposets", len(reposets))
+
     for reposet in tqdm(reposets):
+        # there's at most one reposet per account, so a single transaction is created per account
+        with sentry_sdk.start_transaction(name="precomputer[accounts]"):
```

Comment:
in `__main__` we have
```
    with sentry_sdk.Hub.current.configure_scope() as scope:
        scope.transaction = f"precomputer[{args.command}]"
```
I think we should have same transaction name, else errors generated inside or outside the transaction will have different metadata.

Reply:
We should change `__main__` to replace `with sentry_sdk.Hub.current.configure_scope() as scope:` with
```python
with sentry_sdk.start_transaction(name=f"precomputer[{args.command}]):
```
then change the code here
```suggestion
with sentry_sdk.start_span(op=f"reposet[{reposet.owner_id}]", description=reposet.items):
```

Reply:
ok, so we'll go for a scope per account instead of a transaction per account